### PR TITLE
MAINT: Update core.py with fstrings (issue #15420) 

### PR DESCRIPTION
**Pull request by tohabyuraev**
Replaced parts of code with old string format to fstring for better readability. 
There was only one file in folder "map" 
with old strings. See #15420

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -3272,11 +3269,10 @@ def _scalar_heuristic(arr, elem):
                                 dout._fill_value.flat[0]).all():
                             warnings.warn(
                                 "Upon accessing multidimensional field "
-                                "{indx:s}, need to keep dimensionality "
+                                f"{indx:s}, need to keep dimensionality "
```
Comment (charris) [inline numpy/ma/core.py]: The `:s` looks fishy to me. I think it was fishy to begin with.

File: numpy/ma/core.py [hunk 2]
```diff
@@ -3918,13 +3914,10 @@ def __repr__(self):
                 dtype=str(self.dtype)
             )
             is_structured = bool(self.dtype.names)
-            key = '{}_{}'.format(
-                'long' if is_long else 'short',
-                'flx' if is_structured else 'std'
-            )
+            key = f"{'long' if is_long else 'short'}_{'flx' if is_structured else 'std'}"
```
Comment (charris) [inline numpy/ma/core.py]: OK. Code is a bit obfuscated to start with.

Comment (eric-wieser) [reply numpy/ma/core.py]: I think this was easier to read without the f-strings, since then it could be line-wrapped

File: numpy/ma/core.py (hunk 1 above)
Comment (charris) [reply numpy/ma/core.py]: I'm not convinced this was ever correct.

File: numpy/ma/core.py (hunk 2 above)
Comment (tohabyuraev) [reply numpy/ma/core.py]: Ok, i understood. I thought code in a single line would be more compact and f-string will give readability for programmers.

Comment (charris) [reply numpy/ma/core.py]: The true fix would be to break up the code, but don't do that for this PR :)

**Comment by charris** [comment]
Thanks @tohabyuraev .

**Pull request merged**